### DISCUSSION TITLE
How do you perform a windowed operation on two "layers" in an xarray

### DISCUSSION THREAD

**Discussion by Shaunakde** [pydata/xarray#4782] [Q&A]
I have a Xarray that has 3 dimensions (time, x and y) - they are basically a stack of images. I want to operate in a "windowed" manner on pairs of images. 

Basically take a pair, say the first two "time" images, and call a function on a 5x5 window over them. 

In numpy I would do this by simply slicing appropriately and computing my metric by first forming "patches":

```
params = []
for i in range(0,patch1.shape[0],1):
    for j in range(0,patch1.shape[1],1):
        window1 = np.copy(imga[i:i+N,j:j+N]).flatten()
        window2 = np.copy(imgb[i:i+N,j:j+N]).flatten()
        params.append((window1, window2))
```

Here N is the window size, 5 for instance. Then computing my metric:

```
def f(param):
    return metric_function(*param)

with Pool(4) as p:
    r = list(tqdm.tqdm(p.imap(f,params), total=len(params)))
````

However, I have difficultly translating this to dask/xarray and I would like some help. My first intuition is to use `map_overlay` but I don't know how to deal with the fact that the output would only be the center pixel of the whole patch.

**Comment by ahuang11** [comment]
Maybe related to ds.rolling()
